istioctl/cmd/kubeinject.go

```diff
@@ -167,21 +326,69 @@ func validateFlags() error {
 	return err
 }
 
-var (
-	emitTemplate bool
+func setupKubeInjectParameters(sidecarTemplate *inject.Templates, valuesConfig *string,
+	revision string) (*ExternalInjector, *meshconfig.MeshConfig, error) {
+	var err error
+	var meshConfig *meshconfig.MeshConfig
+	if meshConfigFile != "" {
+		if meshConfig, err = mesh.ReadMeshConfig(meshConfigFile); err != nil {
+			return nil, nil, err
+		}
+	} else {
+		if meshConfig, err = getMeshConfigFromConfigMap(kubeconfig, "kube-inject", revision); err != nil {
+			return nil, nil, err
+		}
+	}
+	injector := &ExternalInjector{nil, nil}
+	if injectConfigFile != "" {
+		injectionConfig, err := ioutil.ReadFile(injectConfigFile) // nolint: vetshadow
+		if err != nil {
+			return nil, nil, err
+		}
+		injectConfig, err := readInjectConfigFile(injectionConfig)
+		if err != nil {
+			return nil, nil, multierror.Append(err, fmt.Errorf("loading --injectConfigFile"))
+		}
+		*sidecarTemplate = injectConfig
+	} else {
+		injector, err = setUpExternalInjector(kubeconfig, revision)
+		if err != nil || injector.clientConfig == nil {
```

Comment:
can injector.clientConfig ever be nil if err == nil? If not, no need to check it. If so, maybe it should return an error in the function?

Reply:
If there is an err, we will fallback to previous method like we discussed in the  issue:
`If there are no webhooks and --injection-url is not set, we will continue to use configmaps and display a deprecation warning`